docs/src/main/asciidoc/mp/integrations/oci.adoc

```diff
@@ -29,80 +29,233 @@ include::{rootdir}/includes/mp.adoc[]
+
+=== Logging
+
+The OCI Logging service is a highly scalable and fully managed single pane of glass for all of the logs in your tenancy. See link:{oci-doc-url}/Content/Logging/home.htm[Logging Overview] in OCI documentation.
+
+Helidon uses the link:{oci-doc-url}/Content/Logging/Concepts/custom_logs.htm[custom logs] feature of OCI Logging for integration. The ability to add custom logs from any OCI Compute instance makes it more flexible to use this feature with other components such as Oracle Kubernetes Engine (OKE).
+
+To enable OCI Logging integration, start by creating a log _group_. Log groups are logical containers for organizing logs. Logs must always be inside log groups. See
+link:{oci-doc-url}/Content/Logging/Task/create-logging-log-group.htm[Create a Log Group] in OCI documentation.
+
+Then, create a custom log object and its associated agent configuration. See link:{oci-doc-url}/Content/Logging/Task/create-logging-log.htm[Creating a Log] in OCI documentation.
+
+Finally, configure Helidon logging to persist its log data to a file that the link:{oci-doc-url}/Content/Logging/Concepts/agent_management.htm[Unified Monitoring Agent] can access and process. For example, in the ``/server/src/main/resources/logging.properties`` file, add:
+
+[source,properties]
+.Defining properties for OCI logging
+----
+handlers=io.helidon.logging.ConsoleHandler,
+java.util.logging.FileHandler
+java.util.logging.FileHandler.pattern=%h/helidon_log/helidon.log
+java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
+----
+
+After you configure OCI logging and launch the application, it will start reporting logs to OCI Logging. Logs are indexed in the system, and searchable through the Console, API, and CLI. You can filter, aggregate, and visualize your logs. For more information, see https://docs.oracle.com/en-us/iaas/Content/Logging/Concepts/searchinglogs.htm[Logging Search] in OCI documentation.
+
+You can also manually push logs to the custom log configuration. First, add the following dependency to the project's ``pom.xml`` file:
+
+[source,xml]
+.Adding the dependency for Logging Ingestion API (to manually push logs)
+----
+<dependency>
+  <groupId>com.oracle.oci.sdk</groupId>
+  <artifactId>oci-java-sdk-loggingingestion</artifactId>
+</dependency>
+----
+
+Then, find the https://docs.oracle.com/en-us/iaas/Content/GSG/Tasks/contactingsupport_topic-Locating_Oracle_Cloud_Infrastructure_IDs.htm#resource-ocid[Oracle Cloud Identifier (OCID)] of the custom log configuration and use that OCID with the https://docs.oracle.com/en-us/iaas/api/#/en/logging-dataplane/20200831/[Logging Ingestion API] from the OCI SDK. See https://docs.oracle.com/en-us/iaas/Content/Logging/Concepts/using_the_api_customlogs.htm#using_the_api_customlogs[Ingesting Custom Logs] in OCI documentation.
+
+=== Tracing
```

Comment:
you might want to also mention using OpenTelemetry for tracing and shipping the traces to OCI APM. see https://medium.com/oracledevs/distributed-tracing-in-helidon-coherence-and-oracle-autonomous-database-with-opentelemetry-and-c60fe965f902